src/Type/MediaItem/Mutation/MediaItemCreate.php

```diff
@@ -0,0 +1,221 @@
+				'mediaItem' => [
+					'type' => Types::post_object( $post_type_object->name ),
+					'resolve' => function( $payload ) {
+						return get_post( $payload['id'] );
+					},
+				],
+			],
+			'mutateAndGetPayload' => function( $input ) use ( $post_type_object, $mutation_name ) {
+
+				/**
+				 * Stop now if a user isn't allowed to upload a mediaItem
+				 */
+				if ( ! current_user_can( 'upload_files' ) ) {
+					throw new \Exception( __( 'Sorry, you are not allowed to upload mediaItems', 'wp-graphql' ) );
+				}
+
+				/**
+				 * If the mediaItem being created is being assigned to another user that's not the current user, make sure
+				 * the current user has permission to edit others mediaItems
+				 */
+				if ( ! empty( $input['authorId'] ) && get_current_user_id() !== $input['authorId'] && ! current_user_can( $post_type_object->cap->edit_others_posts ) ) {
+					throw new \Exception( __( 'Sorry, you are not allowed to create mediaItems as this user', 'wp-graphql' ) );
+				}
+
+				/**
+				 * Set the file name, whether it's a local file or from a URL.
+				 * Then set the url for the uploaded file
+				 */
+				$file_name = basename( $input['filePath'] );
+				$uploaded_file_url = $input['filePath'];
+
+				/**
+				 * Require the file.php file from wp-admin. This file includes the
+				 * download_url and wp_handle_sideload methods
+				 */
+				require_once( ABSPATH . 'wp-admin/includes/file.php' );
+
+				/**
+				 * If the mediaItem file is from a local server, use wp_upload_bits before saving it to the uploads folder
+				 */
+				if ( 'file' === parse_url( $input['filePath'], PHP_URL_SCHEME) ) {
+					$uploaded_file = wp_upload_bits( $file_name, null, file_get_contents( $input['filePath'] ) );
+					$uploaded_file_url = $uploaded_file['url'];
+				}
+
+				/**
+				 * URL data for the mediaItem, timeout value is the default, see:
+				 * https://developer.wordpress.org/reference/functions/download_url/
+				 */
+				$timeout_seconds = 300;
+				$temp_file = download_url( $uploaded_file_url, $timeout_seconds );
+				/**
+				 * Handle the error from download_url if it occurs
+				 */
+				if ( is_wp_error( $temp_file ) ) {
+					throw new \Exception( __( 'Sorry, the URL for this file is invalid, it must be a valid URL', 'wp-graphql' ) );
+				}
+
+				/**
+				 * Build the file data for side loading
+				 */
+				$file_data = [
+					'name'     => $file_name,
+					'type'     => $input['fileType'],
+					'tmp_name' => $temp_file,
+					'error'    => 0,
+					'size'     => filesize( $temp_file ),
+				];
+
+				/**
+				 * Tells WordPress to not look for the POST form fields that would normally be present as
+				 * we downloaded the file from a remote server, so there will be no form fields
+				 * The default is true
+				 */
+				$overrides = [
+					'test_form' => false,
+				];
+
+				/**
+				 * Insert the mediaItem and retrieve it's data
+				 */
+				$file = wp_handle_sideload( $file_data, $overrides );
```

Comment:
not a big deal, but I'd prefer a line break between this and the next code comment block